project-chip/connectedhomeip
# MinMDNS - do not ask for IP addresses for every SRV record that is seen

Various devices will report their SRV records during browse  operations and at boot-up.

General matter nodes however are not interested in these devices unless actively browsing or if explicitly doing operational resolve. This saves the stack trying some A/AAAA queries when a SRV patcket without IP information is received (typically if IP data does not fit in one dnssd packet).

### Changes

- Check if a SRV record is "interesting" before scheduling a IP Address resolution
- #33106 added a test program to validate this (used to be part of this PR however it then got split to minimize PR size)

### Tested

- Used the test program to verify that a IP addresess resolution is not initiated by a running chip-all-clusters app:
   * Run `all-clusters-app`
   * Use wireshark to capture packets on 5353
   * Compile `./scripts/build/build_examples.py --target linux-x64-minmdns build` and run `./out/linux-x64-minmdns/minimal-mdns-tester` to send a SRV packet without IP
   * Wireshark shows IP lookup in previous versions, should show no IP lookup after the update
- Patched the SDK locally to not append IP addresses as additional records and validated that I can still commission chip-all-clusters
   * Modified `QueryResponder.cpp` to update `QueryResponderBase::MarkAditional` to explicitly filter A/AAAA records to not be marked as report additional items, so that they are not included in replies by default as additional items.
